docs/design/auto-storage-scale.md

```diff
@@ -0,0 +1,49 @@
+# Automatic Storage Scaling
+
+# Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+# Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+# High-Level Design
+
+- Create a AutoStorageScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: AutoStorageScaling
+    Metadata:
+	    Name: <storageclusterName>-<deviceClassName>
+	    namespace: <namespace>
+    Spec: 
+	    storageClusterName: <storageclusterName>
+        deviceClassName: <deviceClassName>
+	    capacityLimit: <capacity>   
+    Status:
+	    phase: (waiting/inProgress/completed/failure)
+        LastRunTimeStamp: (only on completed/failure)
+   	    knownOSDSize: 2TiB
+        expectedOSDSize: 4TiB
+        knownOSDCount: 3​ 
+        expectedOSDCount: 3
+    ```
+
+-  Watch prometheus periodically after 10 mins and depending on the OSD fill percentage, recocnile the CR. 
+
+# Low-Level Design 
+
+1) Add a go-routine to watch for Prometheus metrics
+2) Go routine updates the channel and the sync-map with osd percentage
+and also evaluating flag whether resizing is needed
+3) Create a named controller that watcher for channel generic event
+4) If there is an update on the channel reconcile the controller 
+5) Controller will patch the storagecluster with all the device class device set update needed at the same time 
+6) Sync map will act as the desired state and source of truth for the controller.
+
+# Risk
+
+There might cases where system reaches out of memory.
```

Comment:
Do you mean the OSDs will stay in pending state because there is not enough CPU/memory during horizontal scale?

Reply:
Right, in that case we will report failure